lang/src/org/partiql/lang/eval/StrictnessContext.kt

```diff
@@ -0,0 +1,51 @@
+package org.partiql.lang.eval
+
+import org.partiql.lang.ast.MetaContainer
+import org.partiql.lang.errors.ErrorCode
+import org.partiql.lang.errors.PropertyValueMap
+
+
+internal class ErrorDetails(
+    val metas: MetaContainer,
+    val message: String,
+    val errorCode: ErrorCode,
+    val errorContext: PropertyValueMap? = null
+)
+
+internal class StrictnessContext(val strictnessMode: StrictnessMode, val valueFactory: ExprValueFactory) {
+
+    inline fun errorIf(
```

Comment:
API documentation please, intent and exceptions in the `otherwise` thunk is not well defined...

Reply:
Adding...